monkestation/code/modules/reagents/reagent_containers/spray.dm

```diff
@@ -3,3 +3,25 @@
 	volume = 50
 	desc = "A sprayer of advanced Gamer Fuel, designed for rapid deployment during gaming sessions."
 	list_reagents = list(/datum/reagent/consumable/baja_blast = 50)
+
+/obj/item/reagent_containers/spray/chemsprayer/magical
+	name = "Magical Chem Sprayer"
+	desc = "Simply hit the button on the side and this will instantly be filled with a new reagent!"
+	icon_state = "chemsprayer_janitor"
+	item_state = "chemsprayer_janitor"
+	lefthand_file = 'icons/mob/inhands/weapons/guns_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/guns_righthand.dmi'
+	reagent_flags = NONE
+	volume = 1000
+	amount_per_transfer_from_this = 10
+
+/obj/item/reagent_containers/spray/chemsprayer/magical/attack_self(mob/user)
+	cycle_chems(1000)
```

Comment:
Since this is a single use proc, you might as well get rid of the argument here and in the proc itself.